internal/component/pyroscope/scrape/target.go

```diff
@@ -65,9 +65,23 @@ type Target struct {
 func NewTarget(lbls, discoveredLabels labels.Labels, params url.Values) *Target {
 	publicLabels := make(labels.Labels, 0, len(lbls))
 	for _, l := range lbls {
-		if !strings.HasPrefix(l.Name, model.ReservedLabelPrefix) {
-			publicLabels = append(publicLabels, l)
+		if strings.HasPrefix(l.Name, model.ReservedLabelPrefix) {
+			continue
 		}
+
+		// the fact that godeltaprof was used scraping should not be user visible
+		if l.Name == pprofGoDeltaProfMemory {
+			switch l.Value {
+			case pprofGoDeltaProfMemory:
+				l.Value = pprofMemory
+			case pprofGoDeltaProfBlock:
+				l.Value = pprofBlock
+			case pprofGoDeltaProfMutex:
+				l.Value = pprofMutex
+			}
+		}
+
+		publicLabels = append(publicLabels, l)
```

Comment:
Just to sanity check; Is there still the concern that @korniltsev raised in the agent PR about using this with clustering and hash collisions?